### ISSUE TITLE
Cpp test runner empty error message or cryptic errors from tests

### ISSUE THREAD

**Issue by jakubdabek** [exercism/exercism#5743]
I was trying to write `circular-buffer` from cpp track in the online editor and at one point I had this when running tests:
![image](https://user-images.githubusercontent.com/29653525/132057284-3966c3cf-4a56-4a13-9b7e-c4f79d02f168.png)

Then, after writing code to pass one test (I think it was that) I then got this:
![image](https://user-images.githubusercontent.com/29653525/132057378-7bd53ee4-b434-442b-9d44-a4e5dde73bd4.png)
the failure doesn't tell anything about the code.

**Comment by jakubdabek** [comment]
This all happened within an hour as of now, if that helps, username on site is from github

**Comment by jakubdabek** [comment]
I narrowed it down to throwing exceptions when they're not expected, `hello-world` with the following code also exhibits the same behaviour as screenshot 1:
```cpp
string hello()
{
    throw std::runtime_error("what");
}
```

**Comment by jakubdabek** [comment]
And this code probably results in a segfault, but also gives an empty error message:
```cpp
string hello()
{
    return *new(reinterpret_cast<void*>(100ull)) std::string{};
}
```
![image](https://user-images.githubusercontent.com/29653525/132059770-74622b92-1600-458a-9b4d-453c94ebb31f.png)
(Still can't say what causes the behaviour from the second screenshot)

**Comment by iHiD** [comment]
cc @exercism/cpp

**Comment by silvanocerza** [comment]
This is the output coming from [exercism/cpp-test-runner](github.com/exercism/cpp-test-runner) right? @iHiD

**Comment by SleeplessByte** [comment]
@silvanocerza correct. The output file is read and displayed. If it "segfaults" so hard that it fails to write the output file, or writes an empty one, this would be the result.
